### DISCUSSION TITLE
`exec $SHELL` prevents window from closing

### DISCUSSION THREAD

**Discussion by FnControlOption** [ghostty-org/ghostty#3559] [Q&A] [answered]
After editing `~/.zshrc` or `~/.bash_profile`, one way to reload the shell is to run `exec $SHELL`. However, running this command in Ghostty prevents the current window from closing normally.

With Terminal app on macOS:

1. Run `exec $SHELL`
2. Close the window. No pop-up should appear.

With Ghostty:

1. Run `exec $SHELL`
2. Try to close the window. The following pop-up will appear:

> Close Terminal? The terminal still has a running process. If you close the terminal the process will be killed.

<img width="912" alt="Screenshot 2024-12-27 at 11 31 02 AM" src="https://github.com/user-attachments/assets/ca3132d7-e7a2-484d-acf4-b43bacf49620" />

**Comment by 00-kat** [comment, marked as answer]
Ghostty uses [shell integration](https://ghostty.org/docs/config/reference#shell-integration) to determine when you're at a prompt but no program is running.  If you use `exec`, the shell integration injection code won't be run.  Consider manually sourcing the shell integration script in your shell configuration; something like the following would work for zsh:
```sh
if [[ -n $GHOSTTY_RESOURCES_DIR ]]; then
    source "$GHOSTTY_RESOURCES_DIR"/shell-integration/zsh/ghostty-integration
fi
```
(Make sure it's put at the *top* of your configuration)

**Reply by ComplexPlane** [reply]
Can this be automatic to avoid people needing additional configuration? Is it possible to detect the exec'd program is the same as the initial SHELL?

**Reply by 00-kat** [reply]
It probably is, definitely a good feature request although I'm not sure if this one can be repurposed or if you should open a new discussion.
